src/app/view/meter-read/meter-read-table/meter-read-table.component.ts

```diff
@@ -119,8 +125,151 @@ export class MeterReadTableComponent implements OnInit {
+      const startDate = this.formatDateForExport(item.startdate);
+      const endDate = this.formatDateForExport(item.enddate);
+      return `"${startDate}","${endDate}","${item.value}","${item.readtype}"`;
+    });
+
+    // Combine headers and rows
+    const csvContent = [headers.join(','), ...rows].join('\n');
+
+    // Create file and trigger download
+    const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
+    const fileName = `meter_reads_${this.getFormattedDate()}.csv`;
+    saveAs(blob, fileName);
+  }
+
+  async exportToPDF() {
+    if (
+      !this.dataSource ||
+      !this.dataSource.data ||
+      this.dataSource.data.length === 0
+    ) {
+      this.toastrService.warning('No data available to export');
+      return;
+    }
+
+    try {
+      if (!this.pdfMakeLoaded) {
+        const pdfMakeModule = await import('pdfmake/build/pdfmake');
+        const pdfFontsModule = await import('pdfmake/build/vfs_fonts');
+        (pdfMakeModule.default as any).vfs = pdfFontsModule.default.vfs;
+        this.pdfMakeLoaded = true;
+        this.loading = false;
+      }
+      // Data for PDF
+      const tableData = this.dataSource.data.map((item) => {
+        const startDate = this.formatDateForExport(item.startdate);
+        const endDate = this.formatDateForExport(item.enddate);
+        return [startDate, endDate, item.value.toString(), item.readtype];
+      });
+
+      // Insert header row
+      tableData.unshift([
+        'Start Datetime',
+        'End Datetime',
+        'Value(Wh)',
+        'Read Type',
+      ]);
+
+      const headerName = 'Meter Readings';
+
+      const pdfMake = (await import('pdfmake/build/pdfmake')).default;
+
+      // document structure
+      const documentDefinition: TDocumentDefinitions = {
+        content: [
+          { text: headerName, style: 'header' },
+          {
+            text: `Export Date: ${this.datePipe.transform(new Date(), 'MMMM d, yyyy')}`,
+            style: 'subheader',
+          },
+          {
+            style: 'tableExample',
+            table: {
+              headerRows: 1,
+              widths: ['*', '*', 'auto', 'auto'],
+              body: tableData,
+            },
+            layout: {
+              fillColor: (rowIndex: number) => {
+                return rowIndex === 0 ? '#f2f2f2' : null;
+              },
+            },
+          },
+        ],
+        styles: {
+          header: {
+            fontSize: 18,
+            bold: true,
+            margin: [0, 0, 0, 10],
+            color: '#f2be1a',
+          },
+          subheader: {
+            fontSize: 14,
+            bold: true,
+            margin: [0, 10, 0, 5],
+          },
+          tableExample: {
+            margin: [0, 5, 0, 15],
+          },
+        },
+        defaultStyle: {
+          fontSize: 10,
+        },
+      };
+      pdfMake
+        .createPdf(documentDefinition)
+        .download(`meter_reads_${this.getFormattedDate()}.pdf`);
```

Comment:
Please move the pdf generation process into a custom helper function that can be reused in multiple places